lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -4619,6 +4619,12 @@ void SROA_Parameter_HLSL::flattenGlobal(GlobalVariable *GV) {
 
     // Flat Global vector if no dynamic vector indexing.
     bool bFlatVector = !hasDynamicVectorIndexing(EltGV);
+
+    // Disable scalarization of groupshared vector arrays
+    if (GV->getType()->getAddressSpace() == DXIL::kTGSMAddrSpace &&
+        Ty->isArrayTy())
```

Comment:
Struct members are still broken down, just not the vectors, if the top-level global is an array.  This will not change a non-array struct containing arrays of vectors, however.  But that's not a regression, just a limitation that can be fixed in another change.